provider/cmd/pulumi-resource-command/schema.json

```diff
@@ -191,8 +191,8 @@
                     "$ref": "#/types/command:remote:Connection"
                 },
                 "localPath": {
-                    "description": "The path of the file to be copied.",
-                    "type": "string"
+                    "description": "The file/folder to be copied.\nIf the input is an Asset, it will be interperted as the contents of a file.\nIf the input is an Archive, it will be interperted as the contents of a folder.\nIf the input is a string, it will be interpreted as the path to a file or folder.\nBecause assets and archives are understood by the Pulumi model, they use should be\nprefered to raw paths.",
+                    "$ref": "pulumi.json#/Any"
```

Comment:
It might be nicer to offer three separate properties?  `localPath`, `localAsset`, `localArchive`?  Then each could be strongly typed.  They would all have to be optional, and to be mutually exclusive.  

Not 100% on this, but the weak typing that results here is a little unfortunate.

Reply:
I 100% agree that the weak typing is unfortunate. To me, allowing 3 mutually exclusive fields is also weak typing. We allow the type system to accept invalid types. What I should have used (having just remembered this) is `"oneOf"`. This allows us to express a union type. I'll push an update with this.